coalib/bearlib/abstractions/ExternalBearWrap.py

```diff
@@ -0,0 +1,213 @@
+import json
+from functools import partial
+from collections import OrderedDict
+
+from coalib.bears.LocalBear import LocalBear
+from coala_decorators.decorators import enforce_signature
+from coalib.misc.Shell import run_shell_command
+from coalib.results.Diff import Diff
+from coalib.results.Result import Result
+from coalib.results.RESULT_SEVERITY import RESULT_SEVERITY
+from coalib.settings.FunctionMetadata import FunctionMetadata
+
+
+def _prepare_options(options):
+    """
+    Checks for illegal options and raises ValueError.
+
+    :param options:
+        The options dict that contains user/developer inputs.
+    :raises ValueError:
+        Raised when illegal options are specified.
+    """
+    allowed_options = {"executable",
+                       "settings"}
+
+    # Check for illegal superfluous options.
+    superfluous_options = options.keys() - allowed_options
+    if superfluous_options:
+        raise ValueError(
+            "Invalid keyword arguments provided: " +
+            ", ".join(repr(s) for s in sorted(superfluous_options)))
+
+    if not 'settings' in options:
+        options['settings'] = {}
+
+
+def _create_wrapper(klass, options):
+    class ExternalBearWrapBase(LocalBear):
+
+        @staticmethod
+        def create_arguments():
+            """
+            This method has to be implemented by the class that uses
+            the decorator in order to create the arguments needed for
+            the executable.
+            """
+            raise NotImplementedError
+
+        @classmethod
+        def get_executable(cls):
+            """
+            Returns the executable of this class.
+
+            :return:
+                The executable name.
+            """
+            return options["executable"]
+
+        @staticmethod
+        def _normalize_desc(param_entry):
+            """
+            Normalizes the description of the parameters only if there
+            is none provided.
+
+            :param param_entry:
+                A FunctionMetadata OrderedDict entry. It is a
+                tuple, (description, type, default_value). The default_value
+                is used to distinguish optional from non_optional parameters.
+            :return:
+                A value for the OrderedDict in the FunctionMetada object.
+            """
+            if param_entry[0] != "":
+                return param_entry
```

Comment:
looking at DocumentationComment.py I see that it is used to parse the descriptions from the docstring. In my case the descriptions are not in the docstring but rather passed in the decorator.